pkg/metrics/metric/counter.go

```diff
@@ -70,12 +70,53 @@ func NewCounterVec(opts CounterOpts, labelNames []string) DeletableVec[Counter]
 	}
 }
 
+// NewCounterVec creates a new DeletableVec[Counter] based on the provided CounterOpts and
+// partitioned by the given label names.
+func NewCounterVec(opts CounterOpts, labelNames []string) DeletableVec[Counter] {
```

Comment:
Don't think its necessary to wrap the unexported constructor. We should just return the instance of the type.
Returning the interface here says that's all this counter will ever be.